include/drivers/ieee802154/cc1101.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __CC1101_H__
+#define __CC1101_H__
+
+#include <device.h>
+
+/* RF settings
+ *
+ * First 42 entries are for the 42 first registers from
+ * address 0x04 to 0x2D included.
+ * Next, the last 58 entries are for the 58 registers from
+ * extended address 0x00 to 0x39 included
+ *
+ * If CONFIG_IEEE802154_CC1101_RF_PRESET is not used, one will need
+ * no provide 'cc1101_rf_settings' with proper settings. These can
+ * be generated through TI's SmartRF application.
+ *
+ */
+struct cc1101_rf_registers_set {
+	u32_t chan_center_freq0;
+	u16_t channel_limit;
+	/* to fit in u16_t, spacing is a multiple of 100 Hz,
+	 * 12.5KHz for instance will be 125.
+	 */
+	u16_t channel_spacing;
```

Comment:
looks like you don't need neither channel_spacing nor chan_center_freq0. Also you don't have to provide 100 registers unlike in cc1200, so you could revise the attributes and there order.
registers first (with the right number) and channel_limit after.